go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -468,14 +468,39 @@ func (d *DoltSession) NewPendingCommit(ctx *sql.Context, dbName string, roots do
 		return nil, err
 	}
 
+	headCommit := sessionState.headCommit
+	headHash, _ := headCommit.HashOf()
+
 	var mergeParentCommits []*doltdb.Commit
 	if sessionState.WorkingSet.MergeActive() {
 		mergeParentCommits = []*doltdb.Commit{sessionState.WorkingSet.MergeState().Commit()}
+	} else if props.Amend {
+		var parentsHeadForAmend []*doltdb.Commit
+		numParentsHeadForAmend := headCommit.NumParents()
+		for i := 0; i < numParentsHeadForAmend; i++ {
+			parentCommit, err := headCommit.GetParent(ctx, i)
+			if err == nil {
+				parentsHeadForAmend = append(parentsHeadForAmend, parentCommit)
```

Comment:
It doesn't look like we ever read the contents of `parentsHeadForAmend`? Seems like we should be passing it to the commit staging action. 

We should add a test for merge commit handling to make sure we preserve both parents in the new commit; I believe we had a bug with preserving both parents for a merge commit in the initial contribution for the CLI, so would be good to make sure we test that for the SQL path, too.

Reply:
Right you are, this was a bug. Adding the extra tests (specifically, amending a merge commit) surfaced this issue. It's fixed now.